### PR TITLE
Fixed partial print issue on IE / Edge

### PR DESCRIPTION
**Pull request by pixelbits-mk**
Fixed partial print issue for IE / Edge as described in issue #3983.

**Issue Description**
 When printing a PDF from the PDF.js viewer, pages are being cutoff near the bottom of the page. 

**Steps to reproduce** 
1. Open a PDF from PDF.js viewer
2. Click the "Print" button
3. Select a printer and continue to print 
4. On the printed output, observe that the bottom portion of the page is cutoff.

**Fix Description**
 The issue was fixed by changing the rendered size of the canvas to 100%, setting the CSS scale factor to 1 for both the width and height, changing the PRINT_OUTPUT_SCALE to be dependent on DPI, and fixing the PT to PX conversion between the viewport and canvas measurements.

**Notes**
 All tests pass.
 Confirmed that the issue has been fixed on IE 11.494, Edge 25.1, FF 47.0.1, Chrome 51.0
 The PRINT_OUTPUT_SCALE factor has not been changed.

### CODE REVIEW

**Comment by pixelbits-mk** [comment]
Any status on this PR? Please let me know if any changes are requested to move this forward.

**Comment by timvandermeij** [comment]
The commit itself looks good to me. We mainly need to do some more manual testing before merging this.

File: web/pdf_page_view.js [hunk 1]
```diff
@@ -534,26 +534,28 @@ var PDFPageView = (function PDFPageViewClosure() {
...
+      canvas.style.width = '100%';
 
-      var cssScale = 'scale(' + (1 / PRINT_OUTPUT_SCALE) + ', ' +
-                                (1 / PRINT_OUTPUT_SCALE) + ')';
-      CustomStyle.setProp('transform' , canvas, cssScale);
+      CustomStyle.setProp('transform' , canvas, 'scale(1,1)');
       CustomStyle.setProp('transformOrigin' , canvas, '0% 0%');
 
       var canvasWrapper = document.createElement('div');
+      canvasWrapper.style.width = '100%';
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js]: Isn't this actually necessary, given that it should already be set with CSS, see https://github.com/mozilla/pdf.js/blob/336b26a39dc304fc589d0cfb85c52e7d4ee85112/web/viewer.css#L1813?

File: web/pdf_page_view.js [hunk 2]
```diff
@@ -534,26 +534,28 @@ var PDFPageView = (function PDFPageViewClosure() {
...
+      canvasWrapper.style.height = '100%';
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js]: As above, can this not be set using CSS instead?

File: web/pdf_page_view.js [hunk 3]
```diff
@@ -534,26 +534,28 @@ var PDFPageView = (function PDFPageViewClosure() {
...
       var CustomStyle = pdfjsLib.CustomStyle;
       var pdfPage = this.pdfPage;
-
+      // increase viewport or DPI to improve quality
       var viewport = pdfPage.getViewport(1);
       // Use the same hack we use for high dpi displays for printing to get
       // better output until bug 811002 is fixed in FF.
-      var PRINT_OUTPUT_SCALE = 2;
+      var DPI = 144; 
+      var PRINT_OUTPUT_SCALE = DPI/72;
```
Comment (Snuffleupagus) [inline web/pdf_page_view.js]: Nit: To me, this seems kind of orthogonal to the current patch which tries to fix printing cut-off in IE, so I'd suggest just reverting this to `var PRINT_OUTPUT_SCALE = 2;` instead.

**Comment by Snuffleupagus** [comment]
> The commit itself looks good to me.

Generally I agree, but I've added a couple of questions/comments above.

> We mainly need to do more manual testing before merging this.

Agreed; and this is sort of time-consuming to do. We'd need to ensure that this isn't regressing other printing fixes, e.g. PR #7005.
*Note:* Given that we're close to the release of the next Firefox version, we probably don't want to land this *just* before the merge to avoid having to create uplift patches if there're any regressions.

**Comment by yurydelendik** [comment]
The DPI shall be set now to 150%. See #7677. Thanks.
